File: BOJ 2/2535.cpp
#include <cstdio>
#include <algorithm>
#include <string>
#include <vector>
#include <map>
using namespace std;
struct p {
	int nation, number, score;
};
bool compare(const p &a,const p &b) {
	if (a.score != b.score) return a.score > b.score;
}
int main() {
	int n;
	scanf("%d", &n);
	vector<p> v;
	for (int i = 0; i < n; i++) {
		int nation, number, score;
		scanf("%d%d%d", &nation, &number, &score);
		v.push_back({ nation,number,score });
	}
	sort(v.begin(), v.end(), compare);
	int cnt[101] = { 0, };
	int ans = 0;
	for (auto i : v) {
		if (cnt[i.nation] <= 1 && ans<=2) {
			cnt[i.nation]++;
			ans++;
			printf("%d %d\n", i.nation, i.number);
		}
	}
	return 0;
}

File: BOJ(Java 공부용)/2535.java
import java.io.*;
import java.util.ArrayList;
import java.util.List;
import java.util.StringTokenizer;

public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static int[] gox = {0, 1, -1, 0};
    static int[] goy = {1, 0, 0, -1};

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        List<p> v = new ArrayList<>();
        int n = Integer.parseInt(st.nextToken());
        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            int nation = Integer.parseInt(st.nextToken()), num = Integer.parseInt(st.nextToken()), score = Integer.parseInt(st.nextToken());
            v.add(new p(nation, num, score));
        }
        v.sort((a, b) -> Integer.compare(b.score, a.score));
        int[] same = new int[n + 1];
        int cnt = 0;
        for (int i = 0; i < v.size(); i++) {
            int nation = v.get(i).nation, num = v.get(i).num;
            if (cnt == 3) break;
            if (same[nation] >= 2) continue;
            same[nation]++;
            bw.write(nation + " " + num + "\n");
            cnt++;
        }
        bw.flush();
        bw.close();
    }
}

class p {
    int nation, num, score;

    p(int nation, int num, int score) {
        this.nation = nation;
        this.num = num;
        this.score = score;
    }
}